### DISCUSSION TITLE
Easy way to refresh generic_poll_text on mouse click?

### DISCUSSION THREAD

**Discussion by Antonio-Ciociola** [qtile/qtile#2982] [Q&A] [answered]
Is there an easy way to refresh generic_poll_text on mouse click?

**Comment by elParaguayo** [comment, marked as answer]
It's slightly easier now if you've got the latest git version installed as we just merged a change that would help here.

Assuming you do, you could add the following callback
``` python
{"Button1": lazy.widget["genpolltext"].function(lambda w: w.update(w.poll())}
```
If you're not on latest git then try
``` python
def tick_widget():
    qtile.cmd_spawn("qtile cmd-obj -o widget genpolltext -f eval -a 'self.update(self.poll())'")

...

{"Button1": tick_widget}
```

If you have more than one GenPollText widget the you may need to name it (name=...) and use that name instead of "genpolltext".

**Reply by m-col** [reply]
>     qtile.cmd_spawn("qtile cmd-obj -o widget genpolltext -f eval -a 'self.update(self.poll())'")

Interesting - isn't there a better way, perhaps using the IPC server directly in `Qtile`?

**Reply by elParaguayo** [reply]
Well, you can do 
```
qtile.widgets_map["genpolltext"].cmd_eval("self.update(self.poll())")
```
But that's not documented.

As for using the IPC server, that's what `lazy` is doing.

**Reply by m-col** [reply]
I guess you don't even need the `eval` and could use the methods directly. But yeah good point - while `qtile._cmd_spawn` is also undocumented it's the thing I expect is most commonly used.

**Reply by elParaguayo** [reply]
If you didn't use eval, it would need to be a bit longer:
``` python
w = qtile.widgets_map["genpolltext"]
w.update(w.poll())
```
Hmm. Maybe that's shorter...

`cmd_spawn` is undocumented but will be more obvious once we merge the command decorator PR as it will remove that confusion between `qtile.cmd_spawn` and `lazy.spawn`.

**Reply by m-col** [reply]
True that - it will be a great PR
